.github/workflows/build.yml

```diff
@@ -43,14 +45,16 @@ jobs:
     runs-on: ubuntu-latest
     needs: build
     if: github.repository == 'jspecify/jspecify' && github.ref_name == 'main' && github.event_name == 'push'
+    env:
+      JAVA_VERSION: 17
     steps:
       - name: Check out the code
         uses: actions/checkout@v4
       - name: Set up Java
         uses: actions/setup-java@v4
         with:
           distribution: temurin
-          java-version: 17
+          java-version: ${{ env.JAVA_VERSION }}
```

Comment:
actions/setup-java can install several versions at once, so this could be used to install both 17 and 21 (or 22), setting up the latter as the default version, and this would remove the need for:
* the gradle-daemon-jvm.properties toolchainVersion
* the foojay plugin